src/databricks/labs/remorph/reconcile/connectors/oracle_connect.py

```diff
@@ -1,25 +1,90 @@
+import logging
+
+import pyspark
 from pyspark.sql import DataFrame
+from pyspark.sql.functions import expr
 
 from databricks.labs.remorph.reconcile.connectors.source_adapter import SourceAdapter
+from databricks.labs.remorph.reconcile.constants import (
+    Constants,
+    SourceDriver,
+    SourceType,
+)
 from databricks.labs.remorph.reconcile.recon_config import (
     DatabaseConfig,
     Schema,
     Tables,
     TransformRuleMapping,
 )
+from pyspark.errors import PySparkException
 
 
 class OracleAdapter(SourceAdapter):
     def extract_databricks_schema(self, table_conf: Tables, table_name: str) -> list[Schema]:
         pass
 
-    def get_column_list_with_transformation(
-        self, table_conf: Tables, columns: list[str], layer: str
-    ) -> list[TransformRuleMapping]:
-        pass
-
     def extract_schema(self, database_conf: DatabaseConfig, table_conf: Tables) -> list[Schema]:
-        pass
+        try:
+            # get the source table name
+            table_name = table_conf.source_name
+            final_table_name = table_name.split(".")[-1]
+            owner = table_name.split(".")[-2]
+            query = f"""
+            SELECT TABLE_NAME,COLUMN_NAME, DATA_TYPE, DATA_LENGTH, CHAR_LENGTH, DATA_PRECISION, DATA_SCALE, DEFAULT_LENGTH
+            FROM ALL_TAB_COLUMNS
+            WHERE lower(TABLE_NAME) = '{final_table_name.lower()}' and lower(owner) = '{owner.lower()}'
+            """
+            df = (
+                self._get_oracle_reader(query)
+                .load()
+                .withColumn("CHAR_LENGTH", expr("cast(CHAR_LENGTH as int)"))
+                .withColumn("DATA_PRECISION", expr("cast(DATA_PRECISION as int)"))
+                .withColumn("DATA_SCALE", expr("cast(DATA_SCALE as int)"))
+                .withColumn("DATA_LENGTH", expr("cast(DATA_LENGTH as int)"))
+            )
+            df.createOrReplaceTempView("oracle_schema_vw")
+            processed_df = self.spark.sql(
+                """
+                                          select column_name, case when (data_precision is not null and data_scale <> 0) then concat(data_type,"(",data_precision,",",data_scale,")")
```

Comment:
every extract schema function should generate a standard dataframe with
``` SQL
column_name,
data_type,

Use the transform function to translate the out instead of nesting.